uniffi_bindgen/src/interface/enum_.rs

```diff
@@ -92,7 +92,16 @@ use super::{APIConverter, ComponentInterface};
 pub struct Enum {
     pub(super) name: String,
     pub(super) variants: Vec<Variant>,
-    // "Flat" enums do not have variants with associated data.
+    // NOTE: `flat` is a misleading name and to make matters worse, has 2 different
+    // meanings depending on the context :(
+    // * When used as part of Rust scaffolding generation, it means "is this enum
+    //   used with an Error, and that error should we lowered to foreign bindings
+    //   by converting each variant to a string and lowering the variant with that
+    //   string?". In that context, it should probably be called `lowered_as_string` or
+    //   similar.
+    // * When used as part of bindings generation, it means "does this enum have only
+    //   variants with no associated data"? The foreign binding generators are likely
+    //   to generate significantly different versions of the enum based on that value.
```

Comment:
The scaffolding part seems right to me.  I'm not sure that it has a different meaning for the bindings generation though.  The way I think of it, the bindings generation lifts the string lowered by Rust and uses that as the error message, rather than lifting the associated data and storing that.

Reply:
Thanks - I tried to clarify what I meant here, which is that the exact same enum will have a different value for `flat` when used to generate scaffolding than when used to generate bindings.

Reply:
This is really weird indeed.  How does the `flat` value change depending on the context?  Doesn't the same UDL-parsing happen in both cases?

Reply:
Yes and no - for bindings, it goes via uniffi_macros and metadata. In particular, [these semantics](https://github.com/mozilla/uniffi-rs/blob/154739df1867d619ad257d3c02b289c23427fb9f/uniffi_bindgen/src/interface/enum_.rs#L127) don't match what UDL parsing does